shapely/tests/test_predicates.py

```diff
@@ -106,6 +111,53 @@ def test_binary_missing(func):
     assert (~actual).all()
 
 
+@pytest.mark.parametrize("a", all_types)
+@pytest.mark.parametrize("func, func_bin", XY_PREDICATES)
+def test_xy_array(a, func, func_bin):
+    with ignore_invalid(shapely.is_empty(a)):
+        # Empty geometries give 'invalid value encountered' in all predicates
+        # (see https://github.com/libgeos/geos/issues/515)
+        actual = func([a, a], (2, 3))
+        expected = func_bin([a, a], Point(2, 3))
+    assert actual.shape == (2,)
+    assert actual.dtype == np.bool_
+    np.testing.assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize("a", all_types)
+@pytest.mark.parametrize("func, func_bin", XY_PREDICATES)
+def test_xy_array_broadcast(a, func, func_bin):
+    with ignore_invalid(shapely.is_empty(a)):
+        # Empty geometries give 'invalid value encountered' in all predicates
+        # (see https://github.com/libgeos/geos/issues/515)
+        actual = func(a, np.array([[0, 1, 2], [1, 2, 3]]).T)
+        expected = func_bin(a, [Point(0, 1), Point(1, 2), Point(2, 3)])
+    np.testing.assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize("func", [funcs[0] for funcs in XY_PREDICATES])
+def test_xy_with_kwargs(func):
+    out = np.empty((), dtype=np.uint8)
+    actual = func(point, (point.x, point.y), out=out)
+    assert actual is out
+    assert actual.dtype == np.uint8
+
+
+@pytest.mark.parametrize("func", [funcs[0] for funcs in XY_PREDICATES])
+def test_xy_missing(func):
+    actual = func(
+        np.array([point, point, point, None]),
+        np.stack(
```

Comment:
Likewise, this would be easier to read compare in transposed form:

```Python
actual = func(
    [point, point, point, None],
    [[0, 0], [np.nan, 0], [0, np.nan], [0, 0]]
)
```

(wrapping in np.array is not needed)